flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSourceSplitReader.java

```diff
@@ -203,10 +209,21 @@ private class FileStoreRecordIterator implements BulkFormat.RecordIterator<RowDa
                         .orElse(null);
 
         private FileStoreRecordIterator() {
-            this.rowDataSupplier =
-                    valueCountMode
-                            ? new ValueCountRowDataSupplier(this::nextKeyValue)
-                            : new PrimaryKeyRowDataSupplier(this::nextKeyValue);
+            switch (writeMode) {
+                case APPEND_ONLY:
+                    this.rowDataSupplier = new AppendOnlyRowDataSupplier(this::nextKeyValue);
```

Comment:
Looks like we could just reuse the `PrimaryKeyRowDataSupplier` although its key is always a dummy empty key for an append only table. Let me revert this changes.